eglib/src/gmisc-win32.c

```diff
@@ -137,30 +143,40 @@ g_get_home_dir (void)
 const char *
 g_get_user_name (void)
 {
-	const char * retName = g_getenv ("USER");
-	if (!retName)
-		retName = g_getenv ("USERNAME");
-	return retName;
+	if (user_name)
+		return user_name;
+	user_name = g_getenv ("USER");
+	if (!user_name)
+		user_name = g_getenv ("USERNAME");
+	return user_name;
 }
 
-static const char *tmp_dir;
 
 const gchar *
 g_get_tmp_dir (void)
 {
-	if (tmp_dir == NULL){
-		if (tmp_dir == NULL){
-			tmp_dir = g_getenv ("TMPDIR");
-			if (tmp_dir == NULL){
-				tmp_dir = g_getenv ("TMP");
-				if (tmp_dir == NULL){
-					tmp_dir = g_getenv ("TEMP");
-					if (tmp_dir == NULL)
-						tmp_dir = "C:\\temp";
-				}
+	if (tmp_dir == NULL) {
+		tmp_dir = g_getenv ("TMPDIR");
+		if (tmp_dir == NULL) {
+			tmp_dir = g_getenv ("TMP");
+			if (tmp_dir == NULL) {
+				tmp_dir = g_getenv ("TEMP");
+				if (tmp_dir == NULL)
+					tmp_dir = "C:\\temp";
 			}
 		}
 	}
 	return tmp_dir;
 }
 
+void
+g_env_cleanup (void)
+{
+	g_free (user_name);
+	g_free (home_dir);
+	g_free (tmp_dir);
+	tmp_dir = NULL;
```

Comment:
This code will not really work, as tmp_dir at least has one scenario where a constant is assigned (just above).